### PR TITLE
Migrate Counter component to TS.

### PR DESCRIPTION
**Pull request by a11rew**
#### Basic
- [ ] Used plop (`npm run plop`) to create a new component.
- [x] PR has description.
- [x] New component is functional and uses Hooks. 
- [x] Component defines [`PropTypes`](https://reactjs.org/docs/typechecking-with-proptypes.html).
#### Style
- [ ] Styles are added to `NewComponent.modules.scss` file inside of the `NewComponent` folder.
- [ ] Component uses CSS Modules.
- [ ] Design is compatible with [Monday Design System](https://design.monday.com/).
#### Storybook
- [ ] Stories were added to `/src/NewComponent/__stories__/NewComponent.stories.js` file.
- [x] Stories include all flows of using the component.
#### Tests
- [x] Tests are compliant with [TESTING_README.md](TESTING_README.md) instructions.

## Description 
This PR migrates the Counter component and tests to Typescript. Closing #899.

## Converted
[x] Counter.jsx
[x] CounterConstants.js
[x] counter-snapshot-tests.jest.js
[x] counter-tests.jest.js
[x] counter.stories.mdx

### CODE REVIEW

File: src/components/Counter/Counter.tsx [hunk 1]
```diff
@@ -27,13 +61,13 @@ const Counter = ({
   noAnimation
 }) => {
   // Variables
-  const overrideClassName = backwardCompatibilityForProperties([className, wrapperClassName]);
+  const overrideClassName = backwardCompatibilityForProperties([className, wrapperClassName], "") || undefined;
```
Comment (laviomri) [inline src/components/Counter/Counter.tsx:64]: Why was this needed?

**Review by laviomri — COMMENTED** [review]
Thank you for this contribution! 😃  
I left you some comments, please take a look

File: src/components/Counter/__stories__/counter.stories.mdx [hunk 2]
```diff
@@ -38,29 +43,54 @@ export const counterTemplate = createComponentTemplate(Counter);
...
-  "Counters are usually placed after the label of the item they're quantifying, such as the number of notifications. They should only be used to represent a number.",
-  <div className="monday-storybook-counter_usage-link-line">The element the counter refers to should include <Link href="/?path=/docs/popover-tooltip--overview" withoutSpacing>Tooltip,</Link> where necessary, to enhance user understanding. For example, a badge is used in conjunction with an icon.</div>,
-]}/>
 
-<Tip title="Check yourself">Need to indicate information that is not numeric? Use the <Link href="/?path=/docs/data-display-label--overview" size={Link.sizes.SMALL} withoutSpacing>Label</Link> component instead.</Tip>
+<UsageGuidelines
+  guidelines={[
+    "Counters are usually placed after the label of the item they're quantifying, such as the number of notifications. They should only be used to represent a number.",
+    <div className="monday-storybook-counter_usage-link-line">
+      The element the counter refers to should include{" "}
```
[the remark this answers is not in the document]
Comment (a11rew) [reply src/components/Counter/__stories__/counter.stories.mdx:65]: Yes it was auto-format sorry about that 😅

[the remark this answers is not in the document]
Comment (laviomri) [reply src/components/Counter/__stories__/counter.stories.mdx:65]: No problem! It happens to us a lot too :)

File: src/components/Counter/Counter.tsx (hunk 1 above)
Comment (a11rew) [reply src/components/Counter/Counter.tsx]: I have updated this now to return undefined from the `backwardCompatibilityForProperties` function directly as a default value. 

This was needed then because the empty string returned from the `backwardCompatibilityForProperties` call return
 would cause the addition of a redundant `className=""` prop to the output element. By using undefined as the value instead the prop is not added to the element.

File: src/components/Counter/__stories__/counter.stories.mdx [hunk 3]
```diff
@@ -11,14 +11,18 @@ import "./counter.stories.scss";
...
 <Meta
   title="Feedback/Counter"
-  component={ Counter }
-  argTypes={ metaSettings.argTypes }
-  decorators={ metaSettings.decorators }
+  component={Counter}
+  argTypes={{
+    ...metaSettings.argTypes,
+    wrapperClassName: { table: { disable: true } },
+    "data-testid": { table: { disable: true } }
```
[the remark this answers is not in the document]
Comment (a11rew) [reply src/components/Counter/__stories__/counter.stories.mdx:23]: For line 23, the typescript migration guide dictated migrated component props should extend `VibeComponentProps`. That ended up adding `data-testid` as a prop to the component. Storybook picks this up and adds it to the Props section in the docs, also adding a control for it. Since this is not in the existing docs this line is to make Storybook ignore this prop.
https://github.com/mondaycom/monday-ui-react-core/blob/c6519783644b8f21e6f44d6a121b989bd88de679/TYPESCRIPT_MIGRATION.md?plain=1#L11

File: src/components/Counter/Counter.tsx (hunk 1 above)
Comment (laviomri) [reply src/components/Counter/Counter.tsx]: Got it 👍

File: src/components/Counter/__stories__/counter.stories.mdx (hunk 3 above)
[the remark this answers is not in the document]
Comment (laviomri) [reply src/components/Counter/__stories__/counter.stories.mdx:23]: Got it, great 👍

**Comment by a11rew** [comment]
Thank you and thanks for taking the time to review :)

Regarding the other comments you left on the auto-format changes, I reverted them in commit [d5cff3b](https://github.com/mondaycom/monday-ui-react-core/pull/907/commits/20e2970f6005a32ca63c093e2b4cbd0ba972f349) but I think Github would require you to manually mark those conversations as resolved

**Comment by orrgottlieb** [comment]
@a11rew great work! i resolved all of the remaining comments

**Comment by orrgottlieb** [comment]
@a11rew merging and adding the hacktoberfest-approved tag! thank you so much for your contribution
